src/gz/state.c

```diff
@@ -1155,214 +1155,18 @@ void load_state(void *state)
-                      reloc_col_hdr(0x0600E710);
-                      reloc_col_hdr(0x0600F208);
-                      reloc_col_hdr(0x0601167C);
-                      reloc_col_hdr(0x06012508); break;
-        case 0x00F9:  reloc_col_hdr(0x0600075C); break;
-        case 0x0100:  reloc_col_hdr(0x06001438); break;
-        case 0x0112:  reloc_col_hdr(0x06000C98); break;
-        case 0x0113:  reloc_col_hdr(0x06002590);
-                      reloc_col_hdr(0x060038FC); break;
-        case 0x011B:  reloc_col_hdr(0x06000360); break;
-        case 0x011C:  reloc_col_hdr(0x06000578);
-                      reloc_col_hdr(0x06000730); break;
-        case 0x011D:  reloc_col_hdr(0x060003D0); break;
-        case 0x011E:  reloc_col_hdr(0x060005DC); break;
-        case 0x0125:  reloc_col_hdr(0x06007564); break;
-        case 0x0129:  reloc_col_hdr(0x060011B8); break;
-        case 0x0130:  reloc_col_hdr(0x06000DB8); break;
-        case 0x013A:  reloc_col_hdr(0x06000D78); break;
-        case 0x014B:  reloc_col_hdr(0x06000170); break;
-        case 0x014C:  reloc_col_hdr(0x06005CB8);
-                      reloc_col_hdr(0x060091E4); break;
-        case 0x0156:  reloc_col_hdr(0x060011D4); break;
-        case 0x0161:  reloc_col_hdr(0x06000918);
-                      reloc_col_hdr(0x060012C0); break;
-        case 0x0162:  reloc_col_hdr(0x060011EC);
-                      reloc_col_hdr(0x0600238C); break;
-        case 0x0166:  reloc_col_hdr(0x06000908);
-                      reloc_col_hdr(0x06000AF0); break;
-        case 0x016C:  reloc_col_hdr(0x06000D48);
-                      reloc_col_hdr(0x06001430); break;
-        case 0x016F:  reloc_col_hdr(0x06001A58); break;
-        case 0x0170:  reloc_col_hdr(0x06000B70); break;
-        case 0x0178:  reloc_col_hdr(0x06000CB8);
-                      reloc_col_hdr(0x06001B00);
-                      reloc_col_hdr(0x06001C40); break;
-        case 0x0179:  reloc_col_hdr(0x06004618);
-                      reloc_col_hdr(0x0600C080);
-                      reloc_col_hdr(0x0600ECD8); break;
-        case 0x0180:  reloc_col_hdr(0x06001A38);
-                      reloc_col_hdr(0x06003C64); break;
-        case 0x0181:  reloc_col_hdr(0x06001C58);
-                      reloc_col_hdr(0x06001DA8); break;
-        case 0x0185:  reloc_col_hdr(0x06001B70);
-                      reloc_col_hdr(0x06001F70);
-                      reloc_col_hdr(0x06002448);
-                      reloc_col_hdr(0x06002850);
-                      reloc_col_hdr(0x06002D28);
-                      reloc_col_hdr(0x06002FE4);
-                      reloc_col_hdr(0x060033E0);
-                      reloc_col_hdr(0x06003AF0); break;
-        case 0x0189:  reloc_col_hdr(0x0600C2D0); break;
-        case 0x018A:  reloc_col_hdr(0x06000118); break;
-        case 0x0190:  reloc_col_hdr(0x06000B30); break;
+
+      for (int i = 0; i < sizeof(z64_game.col_ctxt.dyn_col) / sizeof(*z64_game.col_ctxt.dyn_col); i++) {
```

Comment:
Looks good, except the collision header relocation loop should probably be lifted out of the object loading loop so as to be performed once after objects are loaded instead of 18 times during object loading monkaS.